modules/distributor/distributor.go

```diff
@@ -396,7 +396,7 @@ func (d *Distributor) checkForRateLimits(tracesSize, spanCount int, userID strin
 }
 
 func (d *Distributor) extractBasicInfo(ctx context.Context, traces ptrace.Traces) (userID string, spanCount, tracesSize int, err error) {
-	user, e := user.ExtractOrgID(ctx)
+	user, e := util.ExtractValidOrgID(ctx)
```

Comment:
I would love to see e2e tests for write path and read path. just to be sure that we don't break it in the future.